cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -151,6 +151,12 @@ pub(crate) enum Inst {
     /// jmpq (reg mem)
     JmpUnknown { target: RegMem },
 
+    XMM_MOV_RM_R {
```

Comment:
Can you add a comment that states for which kind of instructions we'll use this (no need to enumerate precisely which they are, just which kind), and how it differs from XMM_RM_R, please?

Reply:
Hi .. yes, this is to satisfy dependencies w.r.t. the regalloc (maybe regusagecollector?) which needs to understand that we are writing a destination registers and not modifying it (meaning a valid value already existed there).  It was when I was running into runtime errors that I realized the purpose of the _r_r structure was to be able to make this distinction which should apply to moves in general. 
The difference is in the code below in `fn x64_map_regs 

"
        Inst::XMM_MOV_RM_R {
            op: _,
            ref mut src,
            ref mut dst,
        } => {
            src.map_uses(mapper);
            map_def(mapper, dst);
        }
"
Where in XMM_RM_R instead of map_def we have a call to map_mod.

Johnnie

Reply:
Thanks for the explanation; can you add a small code comment that sums it up, above the definition of this enum variant, please?